src/PathMappingRepository.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository;
+
+use ArrayIterator;
+use BadMethodCallException;
+use Puli\Repository\Api\EditableRepository;
+use Puli\Repository\Api\Resource\FilesystemResource;
+use Puli\Repository\Api\Resource\Resource;
+use Puli\Repository\Api\ResourceCollection;
+use Puli\Repository\Api\ResourceNotFoundException;
+use Puli\Repository\Api\UnsupportedResourceException;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Puli\Repository\Resource\GenericResource;
+use Webmozart\Glob\Glob;
+use Webmozart\Glob\Iterator\RegexFilterIterator;
+use Webmozart\PathUtil\Path;
+
+/**
+ * A development path mapping resource repository.
+ * Each resource is resolved at `get()` time to improve
+ * developer experience.
+ *
+ * Resources can be added with the method {@link add()}:
+ *
+ * ```php
+ * use Puli\Repository\PathMappingRepository;
+ *
+ * $repo = new PathMappingRepository();
+ * $repo->add('/css', new DirectoryResource('/path/to/project/res/css'));
+ * ```
+ *
+ * This repository only supports instances of FilesystemResource.
+ *
+ * @since  1.0
+ *
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class PathMappingRepository extends AbstractPathMappingRepository implements EditableRepository
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function get($path)
+    {
+        $path = $this->sanitizePath($path);
+        $resource = $this->resolveResource($path);
```

Comment:
Why couldn't they? `resolveResource` do check first if the resource directly exists in the store (first condition) so as `/foo` will be present there, it will work. I'm pretty sure the tests wouldn't pass if  `$repo->get('/foo')->get('bar') => DirectoryResource` wasn't possible.